cigaradvisor/blocks/article-list/article-list.js

```diff
@@ -1,3 +1,38 @@
+import { readBlockConfig } from '../../scripts/aem.js';
+import { fetchData, getRelativePath } from '../../scripts/scripts.js';
+import { buildArticleTeaser } from '../article-teaser/article-teaser.js';
+
 export default async function decorate(block) {
-  block.innerHTML = '';
+  const configs = readBlockConfig(block);
+  const { category } = configs;
+  const parentDiv = document.createElement('div');
+  parentDiv.classList.add('section');
+  parentDiv.dataset.layout = '50/50';
+  const leftDiv = document.createElement('div');
+  leftDiv.classList.add('left-grid');
+  const rightDiv = document.createElement('div');
+  rightDiv.classList.add('right-grid');
+  let current = rightDiv;
+  if (category) {
+    const articles = await fetchData(getRelativePath(category), '/cigaradvisor/posts/query-index.json', 'category');
+    if (!articles || articles.length === 0) {
+      return;
+    }
+    const articlePromises = [...articles].map(async (article) => {
```

Comment:
This is what i want to avoid. The index should contain everything needed to display any given teaser, there's no need to have a list of promises that need to be awaited.

Reply:
Here, in buildArticleTeaser, we still have to fetch data of each article's author from https://main--famous-smoke-cigaradvisor--hlxsites.hlx.page/cigaradvisor/author/query-index.json. that's making it an async call. Otherwise, articleinfo and categoryinfo will be passed to buildArticleTeaser.

Reply:
Reason, is we don't have all the data for article-teaser under one index. We still have to fetch category/author info for each article.